core/config_api.php

```diff
@@ -727,3 +719,42 @@ function config_is_defined( $p_option, $p_value = null ) {
 	# Value not found in cache
 	return false;
 }
+
+function config_cache_all() {
+	global $g_cache_config, $g_cache_config_access;
+
+	# With oracle databse, ADOdb maps column type "L" to clob.
+	# Because reading clobs is significantly slower, cast them to varchar for faster query execution
+	# Standard max size for varchar is 4000 bytes, so a safe limit is used as 1000 charancters
+	# for multibyte strings (up to 4 bytes per char)
+	if( db_is_oracle() ) {
+		$t_query = 'SELECT config_id, user_id, project_id, type, CAST(value AS VARCHAR(4000)) AS value, access_reqd FROM {config} WHERE dbms_lob.getlength(value)<=1000';
+		$t_result = db_query( $t_query );
+		while( $t_row = db_fetch_array( $t_result ) ) {
+			$t_config = $t_row['config_id'];
+			$t_user = $t_row['user_id'];
+			$t_project = $t_row['project_id'];
+			$g_cache_config[$t_config][$t_user][$t_project] = $t_row['type'] . ';' . $t_row['value'];
+			$g_cache_config_access[$t_config][$t_user][$t_project] = $t_row['access_reqd'];
+		}
```

Comment:
To reduce code duplication, I suggest to move this out of the if and use the same code as other RDBMS, i.e.
```php
if( db_is_oracle() ) {
    $t_query = ...
}
$t_result = ...
while() { ... }
if( db_is_oracle() ) {
    // CLOB query & loop
}
```
And/or maybe use a helper function to assign cache config and access values.

Reply:
i see the code bloat, but:
- i'd rather not add another helper function, only for this usage
- i prefer showing a clear separation of the code paths

lets see how it can be cleaned